cpp/autosar/src/rules/M0-1-10/UnusedSplMemberFunction.ql

```diff
@@ -0,0 +1,32 @@
+/**
+ * @id cpp/autosar/unused-spl-member-function
```

Comment:
Adding a new query requires updating the rule packages metadata file, then running a regeneration script in order to setup the new query.

The full details are described in the development handbook:
https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#step-2-review-and-update-the-rule-package-description-file

But the quick guide would be:
 1. Modify the `cpp/DeadCode.json` file to add a new element to the array for the `queries` property for `M0-1-10`.
 2. Set the metadata appropriately on that query.
 3. Run `python3.9 scripts/generate_rules/generate_package_files.py cpp DeadCode` to regenerate the relevant files.
 4. You will need to update your `isExcluded` predicate to refer to your new query - likely `DeadCodePackage::unusedSplMemberFunctionQuery()` if you keep with the same name.

Doing this will ensure that this query is properly integrated into the deviations mechanism.

Reply:
Addressed in: [7bf12c4](https://github.com/github/codeql-coding-standards/pull/725/commits/7bf12c4902c6acd803b90856b8af6b3c2fff317c)